src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandler.java

```diff
@@ -0,0 +1,84 @@
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service.HomeOfficeInstructService;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service.ListingNotificationHelper;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service.NotificationsHelper;
+
+@Slf4j
+@Component
+public class HearingBundleReadyNotificationHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    protected final HomeOfficeInstructService homeOfficeInstructService;
+    protected final NotificationsHelper notificationsHelper;
+    protected final ListingNotificationHelper listingNotificationHelper;
+
+    public HearingBundleReadyNotificationHandler(
+        HomeOfficeInstructService homeOfficeInstructService,
+        NotificationsHelper notificationsHelper,
+        ListingNotificationHelper listingNotificationHelper) {
+        this.homeOfficeInstructService = homeOfficeInstructService;
+        this.notificationsHelper = notificationsHelper;
+        this.listingNotificationHelper = listingNotificationHelper;
+    }
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && Arrays.asList(
+                        Event.GENERATE_HEARING_BUNDLE,
+                        Event.CUSTOMISE_HEARING_BUNDLE
+                    ).contains(callback.getEvent());
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        log.info("Preparing to send {} notification to HomeOffice for event {}",
+            HEARING_BUNDLE_READY.toString(), callback.getEvent());
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+        final String homeOfficeReferenceNumber = notificationsHelper.getHomeOfficeReference(asylumCase);
+
+        final String caseId = notificationsHelper.getCaseId(asylumCase);
+
+        final HearingInstructMessage hearingInstructMessage
+            = listingNotificationHelper.getHearingBundleReadyInstructMessage(
+                asylumCase, notificationsHelper.getConsumerReference(caseId),
+                notificationsHelper.getMessageHeader(), homeOfficeReferenceNumber);
+
+        log.info("Finished constructing {} notification request for caseId: {}, HomeOffice reference: {}",
+            HEARING_BUNDLE_READY.toString(), caseId, homeOfficeReferenceNumber);
+
+        final String notificationStatus = homeOfficeInstructService.sendNotification(hearingInstructMessage);
+
+        log.info("SENT: {} notification for caseId: {}, HomeOffice reference: {}, status: {}",
```

Comment:
Could you please add the event as well to the logs here for easy traceability?

Reply:
Added event in this log as well